packages/flutter/test/material/text_button_test.dart

```diff
@@ -0,0 +1,996 @@
+    );
+
+    expect(tester.getSize(find.byType(TextButton)), equals(const Size(64.0, 48.0)));
+    expect(tester.getSize(find.byType(Text)), equals(const Size(42.0, 14.0)));
+
+    // textScaleFactor expands text, but not button.
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Material(
+          child: MediaQuery(
+            data: const MediaQueryData(textScaleFactor: 1.3),
+            child: Center(
+              child: TextButton(
+                onPressed: () { },
+                child: const Text('ABC'),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // Scaled text rendering is different on Linux and Mac by one pixel.
+    // TODO(gspencergoog): Figure out why this is, and fix it. https://github.com/flutter/flutter/issues/12357
+    expect(tester.getSize(find.byType(TextButton)).width, isIn(<double>[70.0, 71.0]));
+    expect(tester.getSize(find.byType(TextButton)).height, isIn(<double>[47.0, 48.0]));
+    expect(tester.getSize(find.byType(Text)).width, isIn(<double>[54.0, 55.0]));
+    expect(tester.getSize(find.byType(Text)).height, isIn(<double>[18.0, 19.0]));
+
+    // Set text scale large enough to expand text and button.
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: Material(
+          child: MediaQuery(
+            data: const MediaQueryData(textScaleFactor: 3.0),
+            child: Center(
+              child: TextButton(
+                onPressed: () { },
+                child: const Text('ABC'),
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    // Scaled text rendering is different on Linux and Mac by one pixel.
+    // TODO(gspencergoog): Figure out why this is, and fix it. https://github.com/flutter/flutter/issues/12357
+    expect(tester.getSize(find.byType(TextButton)).width, isIn(<double>[133.0, 134.0]));
+    expect(tester.getSize(find.byType(TextButton)).height, equals(48.0));
+    expect(tester.getSize(find.byType(Text)).width, isIn(<double>[126.0, 127.0]));
+    expect(tester.getSize(find.byType(Text)).height, equals(42.0));
+  });
+
+
+  testWidgets('TextButton size is configurable by ThemeData.materialTapTargetSize', (WidgetTester tester) async {
+    final Key key1 = UniqueKey();
+    await tester.pumpWidget(
+      Theme(
+        data: ThemeData(materialTapTargetSize: MaterialTapTargetSize.padded),
+        child: Directionality(
+          textDirection: TextDirection.ltr,
+          child: Material(
+            child: Center(
+              child: TextButton(
+                key: key1,
+                child: const SizedBox(width: 50.0, height: 8.0),
+                onPressed: () { },
+              ),
+            ),
+          ),
+        ),
+      ),
+    );
+
+    expect(tester.getSize(find.byKey(key1)), const Size(66.0, 48.0));
+
+    final Key key2 = UniqueKey();
+    await tester.pumpWidget(
```

Comment:
Same comment as ContainedButton's test, could put this into an inner function to avoid duplicating

Reply:
Yes, done.